pyccl/ccl_background.i

```diff
@@ -5,103 +5,124 @@
 %}
 
 // Enable vectorised arguments for arrays
-%apply (double* IN_ARRAY1, int DIM1) {(double* a, int na)};
-%apply (double* IN_ARRAY1, int DIM1) {(double* chi, int nchi)};
-%apply (double* ARGOUT_ARRAY1, int DIM1) {(double* output, int nout)};
+%apply (double *IN_ARRAY1, int DIM1) {(double *a, int na)};
+%apply (double *IN_ARRAY1, int DIM1) {(double *chi, int nchi)};
+%apply (int DIM1, double *ARGOUT_ARRAY1) {(int nout, double *output)};
```

Comment:
I have to say, I'm not a fan of this style change.  I'm not usually one to impose style preferences, but it doesn't seem like a great idea to change a bunch of existing code to a different (and imo worse) style.

To me `double* a` is significantly more legible than `double *a`.  `double*` is the type - pointer to a double.  So I always find the original syntax to be clearer about what the code means.  I.e. `a` is a pointer to double.  Rather than `a` when dereferenced (`*a`) is a double, which is kind of what the latter syntax means.  

It ends up in the same place (as it has to, since white space is not significant in C), but the former syntax more direct to me.

However, it's probably worth hearing from some other CCL coders about this to settle on a consistent style.  Both forms were in the existing code, and it is probably worth being consistent at least, rather than having both styles interspersed.

Reply:
Happy to switch back the location of the `*`.

I swapped the order of the size and pointer in the SWIG apply statements for good reason. SWIG by default for output arrays keeps only the first of the two parameter names and expects you to pass in the size of the output. Thus when we list the size first, it generates a function prototype in python that looks sensible.